lib/rbs_rails/active_record.rb

```diff
@@ -140,6 +141,24 @@ def #{singular_name}_ids=: (::Array[::Integer]) -> ::Array[::Integer]
         end.join("\n")
       end
 
+      private def has_and_belongs_to_many
+        klass.reflect_on_all_associations(:has_and_belongs_to_many).map do |a|
+          @dependencies << a.klass.name
+
+          singular_name = a.name.to_s.singularize
+          type = Util.module_name(a.klass)
+          collection_type = "#{type}::ActiveRecord_Associations_CollectionProxy"
+          @dependencies << collection_type
+
+          <<~RUBY.chomp
+            def #{a.name}: () -> #{collection_type}
+            def #{a.name}=: (#{collection_type} | ::Array[#{type}]) -> (#{collection_type} | ::Array[#{type}])
+            def #{singular_name}_ids: () -> ::Array[::Integer]
+            def #{singular_name}_ids=: (::Array[::Integer]) -> ::Array[::Integer]
+          RUBY
+        end.join("\n")
+      end
```

Comment:
Note: I duplicated `#has_many` method and only modified the argument of `klass.reflect_on_all_associations`.